react-components/src/hooks/useSearchAssetsMappedPointCloudAnnotations.ts

```diff
@@ -0,0 +1,133 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+
+import { type AddModelOptions } from '@cognite/reveal';
+import {
+  type AnnotationFilterProps,
+  type AnnotationsBoundingVolume,
+  type Asset,
+  type CogniteClient,
+  type AnnotationModel
+} from '@cognite/sdk';
+import { useQuery, type UseQueryResult } from '@tanstack/react-query';
+import { chunk, uniqBy } from 'lodash';
+
+export const useAllAssetsMappedPointCloudAnnotations = (
+  sdk: CogniteClient,
+  models: AddModelOptions[]
+): UseQueryResult<Asset[]> => {
+  return useQuery(
+    ['reveal', 'react-components', 'all-assets-mapped-point-cloud-annotations', models],
+    async () => {
+      const assetMappings = await getAssetsMappedPointCloudAnnotations(sdk, models);
+      return assetMappings;
+    },
+    {
+      staleTime: Infinity
+    }
+  );
+};
+
+export const useSearchAssetsMappedPointCloudAnnotations = (
+  models: AddModelOptions[],
+  sdk: CogniteClient,
+  query: string
+): UseQueryResult<Asset[]> => {
+  const { data: assetMappings, isFetched } = useAllAssetsMappedPointCloudAnnotations(sdk, models);
+
+  return useQuery(
+    ['reveal', 'react-components', 'search-assets-mapped-point-cloud-annotations', query, models],
+    async () => {
+      if (query === '') {
+        return assetMappings;
+      }
+
+      const filteredSearchedAssets =
+        assetMappings?.filter((asset) => {
+          const isInName = asset.name.toLowerCase().includes(query.toLowerCase());
+          const isInDescription = asset.description?.toLowerCase().includes(query.toLowerCase());
+
+          return isInName || isInDescription;
+        }) ?? [];
+
+      return filteredSearchedAssets;
+    },
+    {
+      staleTime: Infinity,
+      enabled: isFetched && assetMappings !== undefined
+    }
+  );
+};
+
+async function getAssetsMappedPointCloudAnnotations(
+  sdk: CogniteClient,
+  models: AddModelOptions[]
+): Promise<Asset[]> {
+  const modelIdList = models.map((model) => model.modelId);
+  const pointCloudAnnotations = await getPointCloudAnnotations(modelIdList, sdk);
+  const result = await getPointCloudAnnotationAssets(pointCloudAnnotations, sdk);
+
+  return result;
+}
+
+async function getPointCloudAnnotationAssets(
+  pointCloudAnnotations: AnnotationModel[],
+  sdk: CogniteClient
+): Promise<Asset[]> {
+  const annotationMapping = pointCloudAnnotations.map((annotation) => ({
+    assetId:
+      (annotation.data as AnnotationsBoundingVolume).assetRef?.id ??
+      (annotation.data as AnnotationsBoundingVolume).assetRef?.externalId ??
+      annotation.id ??
+      ''
```

Comment:
Can we remove these two lines? I think it makes more sense to check for `undefined` in the below filter, rather than empty string

Reply:
filtering based on `undefined` doesn't seems to work here, can be something to do with typeguard, will check and let you know